boards/arc/nsim/nsim_em11d.yaml

```diff
@@ -0,0 +1,11 @@
+identifier: nsim_em11d
+name: EM11D Nsim simulator
+type: sim
+simulation: nsim
+arch: arc
+toolchain:
+  - arcmwdt
```

Comment:
I'd propose to not limit the platform to `arcmwdt`, instead let's limit the `test_calculation` dsp testcase to `arcmwdt`.

The rest of the thing can be build/run with GNU toolchain. 
Of course such variant (`nsim_em11d` + GNU toolchain) is currently mostly for testing and not for any real use cases.

To implement this we probably need to add EM GNU toolchain options and keep `EM11D_DSP_OPTIONS` empty for GNU toolchain.

Reply:
For this test, nsim_em11d + GNU toolchain doesn't need EM GNU toolchain options, as there is no DSP operation in `test_load_store`. Now only enabled `EM11D_DSP_OPTIONS` for mwdt toolchain.